stan/math/prim/arr/err/is_matching_size.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef STAN_MATH_PRIM_ARR_ERR_IS_MATCHING_SIZE_HPP
+#define STAN_MATH_PRIM_ARR_ERR_IS_MATCHING_SIZE_HPP
+
+#include <stan/math/prim/scal/err/is_size_match.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Check if two structures are the same size.
+ *
+ * This function checks the runtime sizes for
+ * variables that implement a <code>size()</code>
+ * method.
+ *
+ * @tparam T_y1 Type of the first variable
```

Comment:
We should mention here `T_y1` type must have a `.size()` function (like the comment right above).

So... maybe something like:
"@tparam T_y1 Type of the first variable. This variable must have a <code>size()</code> method defined"

If you have a better way to describe that, it'll work too.